mne/time_frequency/tfr.py

```diff
@@ -1107,9 +1162,225 @@ def plot(self, picks, baseline=None, mode='mean', tmin=None, tmax=None,
+        import matplotlib.pyplot as plt
+
+        if topomap_args is None:
+            topomap_args = dict()
+
+        cmap = _setup_cmap(cmap)
+
+        # prepare axes for topomap
+        fig, tf_ax, map_ax, cbar_ax = _prepare_joint_axes(len(timefreqs)
+                                                          if timefreqs is
+                                                          not None else 1)
+
+        _, tf_data, tf_times, tf_freqs = \
+            self._plot(picks=picks, baseline=baseline, mode=mode, tmin=tmin,
+                       tmax=tmax, fmin=fmin, fmax=fmax, vmin=vmin, vmax=vmax,
+                       cmap=cmap, dB=dB, colorbar=False, show=show,
+                       title=title, axes=tf_ax, layout=layout, yscale=yscale,
+                       aggregate=aggregate, exclude=exclude, verbose=verbose)
+
+        if timefreqs is None:
+            # find the maximum peak
+            from scipy.signal import argrelmax
+            order = tf_data.shape[2] // 30
+            if order < 1:
+                order = 1
+            peaks_idx = argrelmax(tf_data, order=order, axis=2)
+            if peaks_idx[0].size == 0:
+                _, p_t, p_f = np.unravel_index(tf_data.argmax(),
+                                               tf_data.shape)
+                timefreqs = [(tf_times[p_t], tf_freqs[p_f])]
+            else:
+                peaks = [tf_data[0, f, t] for f, t in
+                         zip(peaks_idx[1], peaks_idx[2])]
+                peakmax_idx = np.argmax(peaks)
+                peakmax_time = tf_times[peaks_idx[2][peakmax_idx]]
+                peakmax_freq = tf_freqs[peaks_idx[1][peakmax_idx]]
+
+                timefreqs = [(peakmax_time, peakmax_freq)]
+
+        timefreqs = {k: np.asarray(timefreqs[k])
+                     if isinstance(timefreqs, dict) else np.array([0, 0])
+                     for k in timefreqs}
+
+        vmin, vmax = _setup_vmin_vmax(tf_data, vmin, vmax)
+
+        # topomap
+        contours = topomap_args.get('contours', 6)
+        locator, contours = _set_contour_locator(vmin, vmax, contours)
+
+        topomap_args_pass = {k: v for k, v in topomap_args.items() if
+                             k not in ('axes', 'show', 'colorbar')}
+        topomap_args_pass['outlines'] = (topomap_args['outlines'] if 'outlines'
+                                         in topomap_args else 'skirt')
+        topomap_args_pass['contours'] = contours
+
+        for sub_map_ax, ((time, freq), avg) in zip(map_ax, timefreqs.items()):
+            time_half_range, freq_half_range = avg / 2.
+
+            if time_half_range == 0:
+                time = self.times[np.argmin(np.abs(self.times - time))]
+            if freq_half_range == 0:
+                freq = self.freqs[np.argmin(np.abs(self.freqs - freq))]
+
+            tmin, tmax = time - time_half_range, time + time_half_range
+            fmin, fmax = freq - freq_half_range, freq + freq_half_range
```

Comment:
I guess it could also just be directly evaluated in the call to `plot_topomap`. What do you think?